addon/components/paper-switch.js

```diff
@@ -25,12 +25,35 @@ export default BaseFocusable.extend(RippleMixin, ProxiableMixin, ColorMixin, {
   dragAmount: null,
   switchWidth: null,
 
-  onDidInsertElement: Ember.on('didInsertElement', function() {
-    // Don't set up anything if the switch is disabled
-    if (this.get('disabled')) { return; }
+  didInsertElement() {
+    this._super(...arguments);
+    //Only setup if the switch is not disabled
+    if (!this.getAttr('disabled')) {
+      this._setupSwitch();
+    }
+  },
 
-    this._super();
+  didInitAttrs() {
+    this._super(...arguments);
+    assert('{{paper-switch}} requires an `onchange` function', this.getAttr('onchange') && typeof this.getAttr('onchange') === 'function');
+  },
+
+  willDestroyElement() {
+    this._super(...arguments);
+
+    if (this.switchHammer) {
+      this.switchHammer.destroy();
+    }
+    if (this.thumbElementHammer) {
+      this.switchHammer.destroy();
+    }
+  },
 
+  disabledDidChange: observer('disabled', function() {
+    this._setupSwitch();
+  }),
```

Comment:
Let's use the `didReceiveAttrs` instead of this observer.
More info: https://guides.emberjs.com/v2.2.0/components/the-component-lifecycle/#toc_formatting-component-attributes-with-code-didreceiveattrs-code

> Since the didReceiveAttrs hook is called every time a component's attributes are updated whether on render or re-render, you can use the hook to effectively **act as an observer**, ensuring code is executed every time an attribute changes.